axelor-marketing/src/main/resources/views/TargetList.xml

```diff
@@ -27,11 +27,17 @@
 		  	<field name="partnerQuery" colSpan="12" showIf="partnerQueryTypeSelect == 1"/>
 		  	<field name="partnerFilterList" colSpan="12" layout="table" showIf="partnerQueryTypeSelect == 0">
 		  	 	<editor x-show-titles="false">
-	            <field name="metaField" onSelect="action-target-list-partner-filters-field-domain" colSpan="3" widget="SuggestBox" required="true" onChange="action-filter-method-update-target-field" target-name="label"/>
-	            <field name="targetField" colSpan="3" showIf="metaField.relationship != null" requiredIf="metaField.relationship != null" onChange="action-filter-method-update-target-type"/>
-	            <field name="operator"  colSpan="3" required="true"/>
-	            <field name="value" colSpan="3" onChange="action-studio-filter-validate-values"/>
+	            <field name="metaField" readonlyIf="targetType != null &amp;&amp; targetType != 'ManyToOne' &amp;&amp; targetType != 'ManyToMany' &amp;&amp; targetType != 'OneToOne'" onSelect="action-target-list-partner-filters-field-domain" colSpan="3" widget="SuggestBox" required="true" onChange="action-filter-method-update-target-field" target-name="label"/>
```

Comment:
Don't change domain on metaField but create new dummy 'targetMetaField'. Display it when target type is m2o/m2m. 
User will use that field to select targetValue. Put it under  targetField.